golang/cosmos/x/vtransfer/types/baseaddr.go

```diff
@@ -0,0 +1,41 @@
+package types
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func trimSlashPrefix(s string) string {
+	return strings.TrimPrefix(s, "/")
+}
+
+// ExtractBaseAddress extracts the base address from a parameterized address.
+func ExtractBaseAddress(addr string) (string, error) {
+	parsed, err := url.Parse(addr)
+	if err != nil {
+		return "", err
+	}
+
+	// Remove leading slashes from the path and opaque fields so that only
+	// parsed relative paths match the expected paths.
+	expected := url.URL{
+		Opaque:      trimSlashPrefix(parsed.Opaque),
```

Comment:
I think we want to definitively assert an empty value for `Opaque` until such time (if ever) that we commit to "orch:".
```suggestion
```